lib/ProductOpener/GS1.pm

```diff
@@ -1461,7 +1588,56 @@ sub read_gs1_json_file($$) {
 	# which is the expected format of the ProductOpener::GS1 module
 	convert_single_text_property_to_direct_value($json_ref);
 		
-	convert_gs1_json_message_to_off_products_csv($json_ref, $products_ref);
+	convert_gs1_json_message_to_off_products_csv($json_ref, $products_ref, $messages_ref);
+}
+
+
+sub generate_gs1_message_identifier() {
+
+	# local GLN + 60 random hexadecimal characters
+	my $identifier = deep_get(\%options, qw(gs1 local_gln)) . "_";
+	$identifier .= sprintf("%x", rand 16) for 1..60;
+
+	return $identifier;
+}
+
+
+sub generate_gs1_confirmation_message($$) {
```

Comment:
A doc would be good, it's already quite complicated this gs1 stuff, so better document !